pkg/ingester/ingester.go

```diff
@@ -248,15 +313,24 @@ func (i *Ingester) Push(ctx old_ctx.Context, req *client.WriteRequest) (*client.
 			return nil, wrapWithUser(err, userID)
 		}
 	}
-	client.ReuseSlice(req.Timeseries)
+	defer client.ReuseSlice(req.Timeseries)
```

Comment:
Pointing https://github.com/cortexproject/cortex/pull/2000 here as it also includes the same fix, and there is discussion going on for the fix. I have found WAL to cause panics without that fix, so maybe we want to wait for some conclusion on that PR (merging this PR without this fix would make it unsafe to deploy WAL).

Reply:
It has been merged and I have rebased to remove the change from this PR.